### PR TITLE
Gcw 2752 add preferred contact number field on user form

### PR DESCRIPTION
**Pull request by akshaygwd**
Hi Team,

This PR contains Addition of a preferred contact number field in the stock app for new user form,
It also contains the UI update for the same new user form.

Please review.
Thanks

### CODE REVIEW

File: app/controllers/users/new.js [hunk 1]
```diff
@@ -22,55 +22,83 @@ export default Ember.Controller.extend({
...
     this.set("position", "");
+    this.set("preferredPhone", "");
   },
 
-  formatMobileNumber(){
-    var mobile = this.get('mobilePhone');
-    if(mobile.length){
-      return config.APP.HK_COUNTRY_CODE + this.get('mobilePhone');
+  formatMobileNumber(mobileNumber) {
+    if (mobileNumber.length) {
```
Comment (Darshan-Chauhan) [inline app/controllers/users/new.js]: I dont understand use case of this function.
We're passing a number and returning the same number in if and else both conditions. Do we even require this function?

File: app/controllers/users/new.js [hunk 2]
```diff
@@ -22,55 +22,83 @@ export default Ember.Controller.extend({
...
   },
 
   actions: {
     saveUser() {
-      var loadingView = getOwner(this).lookup('component:loading').append();
-      var mobilePhone = this.formatMobileNumber();
-      var firstName = this.get('firstName');
-      var lastName = this.get('lastName');
-      var organisationId = this.get('organisationId');
+      var loadingView = getOwner(this)
```
Comment (Darshan-Chauhan) [inline app/controllers/users/new.js]: I think all `var` declarations can be replaced with `const.

Comment (akshaygwd) [reply app/controllers/users/new.js]: @Darshan-Chauhan Please review the PR I did the suggested changes

File: app/controllers/users/new.js [hunk 3]
```diff
@@ -25,28 +25,21 @@ export default Ember.Controller.extend({
...
-      var lastName = this.get("lastName");
-      var organisationId = this.get("organisationId");
-      var position = this.get("position");
-      var email = this.get("email");
+      const mobilePhone =
+        this.get("preferredPhone.length") &&
+        `${config.APP.HK_COUNTRY_CODE}${this.get("mobilePhone")}`;
+      const preferredPhone =
+        this.get("preferredPhone.length") && this.get("preferredPhone");
+      const firstName = this.get("firstName");
```
Comment (Darshan-Chauhan) [inline app/controllers/users/new.js]: this.get('preferredPhone') can be cached

File: app/controllers/users/new.js [hunk 4]
```diff
@@ -25,28 +25,21 @@ export default Ember.Controller.extend({
...
-        config.APP.HK_COUNTRY_CODE +
-        this.formatMobileNumber(this.get("mobilePhone"));
-      var preferredPhone = this.formatMobileNumber(this.get("preferredPhone"));
-      var firstName = this.get("firstName");
-      var lastName = this.get("lastName");
-      var organisationId = this.get("organisationId");
-      var position = this.get("position");
-      var email = this.get("email");
+      const mobilePhone =
+        this.get("preferredPhone.length") &&
```
Comment (Darshan-Chauhan) [inline app/controllers/users/new.js]: I think this should be this.get('mobilePhone.length').

**Pull request merged**